### ISSUE TITLE
Generated SRA submission issues when sending to ENA

### ISSUE THREAD

**Issue by djcomlab** [ISA-tools/isa-api#158]
COPO have reported a few issues that are blockers for them, when submitting SRA XML to ENA.

* ``sra_broker_inform_on_status`` and ``sra_broker_inform_on_error`` supplied in ``sra_settings`` not reflected in the generated xmls
* Stand-alone samples (i.e. those not linked to submitted data files) are not reflected in ``sample_set.xml``
* Samples attached to more than one datafile, repeated in ``sample_set.xml``

Example receipt from ENA's submission web service: 
```
<?xml version="1.0" encoding="UTF-8"?>
<?xml-stylesheet type="text/xsl" href="receipt.xsl"?>
<RECEIPT receiptDate="2016-11-01T11:56:15.391Z" submissionFile="submission.xml" success="false"><SUBMISSION alias="58187af3d127fd3ee0a0f9e2"/><MESSAGES>
<ERROR> Project XML  is missing or invalid or not included in submission XML</ERROR>
<ERROR>The sample alias (58187af3d127fd3ee0a0f9e2:sample:1EX) found more than once but sample alias must be unique.</ERROR>
<ERROR>The sample alias (58187af3d127fd3ee0a0f9e2:sample:1EX) found more than once but sample alias must be unique.</ERROR>
<INFO> ADD action for the following XML:  sample_set.xml experiment_set.xml run_set.xml     project_set.xml </INFO>
<INFO>Inform_on_error is not filled in; auto populated from Submission account. </INFO>
</MESSAGES>
<ACTIONS>ADD</ACTIONS>
<ACTIONS>ADD</ACTIONS>
<ACTIONS>ADD</ACTIONS>
<ACTIONS>ADD</ACTIONS>
</RECEIPT>
```

using SRA settings JSON:
```
 {
 "sra_center": "EI",
 "sra_broker": "EI",
 "sra_action": "ADD",
 "sra_broker_inform_on_status": "support@.....org",
 "sra_broker_inform_on_error": "support@......org"
}
```
and ISA JSON: [isa_json.json.txt](https://github.com/ISA-tools/isa-api/files/564002/isa_json.json.txt)

**Comment by djcomlab** [comment]
using ``json2sra.convert2()``

**Comment by djcomlab** [comment]
The exporter currently assumes 1-1 sample-data file relationship, which is probably what's causing points 2 and 3.

**Comment by djcomlab** [comment]
On further inspection, found that sample was being written into ``sample_set.xml`` per assay to export.

**Comment by djcomlab** [comment]
Missing ``sra_broker_inform_on_status`` and ``sra_broker_inform_on_error`` is because code currently extracts this info from ``comments``, where we are deprecating embedding SRA contacts in ``comments`` and replacing with external JSON settings for SRA submissions.

**Comment by djcomlab** [comment]
Contacted datasubs@ebi.ac.uk for assistance, as I cannot see the source of the Project XML error.

**Comment by djcomlab** [comment]
ENA devs have opened a ticket ENA-1492 on their end to fix.

**Comment by djcomlab** [comment]
Fixed released in v0.3.3, closing.
